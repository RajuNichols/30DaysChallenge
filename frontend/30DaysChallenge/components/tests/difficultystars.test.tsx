import React from 'react';
import renderer, { ReactTestRenderer, ReactTestInstance } from 'react-test-renderer';
import DifficultyStars from '../difficultystars';

describe('DifficultyStars', () => {
  let testRenderer: ReactTestRenderer;
  let testInstance: ReactTestInstance;

  afterEach(() => {
    testRenderer.unmount();
  });

  it('renders the correct number of solid and blank stars based on difficulty', () => {
    const difficulty = 3;
    testRenderer = renderer.create(<DifficultyStars difficulty={difficulty} />);
    const tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct star images based on size prop', () => {
    const difficulty = 3;
    const size = 1;

    testRenderer = renderer.create(<DifficultyStars difficulty={difficulty} size={size} />);
    const tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Add more tests as needed
});