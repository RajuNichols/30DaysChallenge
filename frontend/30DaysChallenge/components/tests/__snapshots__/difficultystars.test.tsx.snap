// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DifficultyStars renders the correct number of solid and blank stars based on difficulty 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Image
    source={1}
    style={Object {}}
  />
  <Image
    source={1}
    style={Object {}}
  />
  <Image
    source={1}
    style={Object {}}
  />
  <Image
    source={1}
    style={Object {}}
  />
  <Image
    source={1}
    style={Object {}}
  />
</View>
`;

exports[`DifficultyStars renders the correct star images based on size prop 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Image
    source={1}
    style={Object {}}
  />
  <Image
    source={1}
    style={Object {}}
  />
  <Image
    source={1}
    style={Object {}}
  />
  <Image
    source={1}
    style={Object {}}
  />
  <Image
    source={1}
    style={Object {}}
  />
</View>
`;
